File: app/src/test/java/blue/koenig/kingsfinances/TestExpensesSubject.java
package blue.koenig.kingsfinances;

import com.koenig.commonModel.finance.CostDistribution;
import com.koenig.commonModel.finance.Expenses;

import blue.koenig.kingsfinances.model.calculation.StatisticEntry;

import static blue.koenig.kingsfinances.TestHelper.thomas;

/**
 * Created by Thomas on 02.01.2018.
 */

public class TestExpensesSubject extends TestSubject<Expenses> {
    public void updateDebts(StatisticEntry oldStatisticEntry, StatisticEntry newStatisticEntry) {
        update(makeExpensesFromDebts(oldStatisticEntry), makeExpensesFromDebts(newStatisticEntry));
    }

    private Expenses makeExpensesFromDebts(StatisticEntry statisticEntry) {
        CostDistribution costDistribution = TestHelper.makeCostDistribution(statisticEntry.getEntryFor(thomas), 0, 0, statisticEntry.getEntryFor(thomas));
        return new Expenses("", "", "", costDistribution.sumReal(), costDistribution, statisticEntry.getDate(), "");
    }
}

File: app/src/test/java/blue/koenig/kingsfinances/TestExpensesSubject.kt
package blue.koenig.kingsfinances

import com.koenig.commonModel.finance.Expenses
import com.koenig.commonModel.finance.statistics.StatisticEntryDeprecated

/**
 * Created by Thomas on 02.01.2018.
 */

class TestExpensesSubject : TestSubject<Expenses>() {
    fun updateDebts(oldStatisticEntry: StatisticEntryDeprecated, newStatisticEntry: StatisticEntryDeprecated) {
        update(makeExpensesFromDebts(oldStatisticEntry), makeExpensesFromDebts(newStatisticEntry))
    }

    private fun makeExpensesFromDebts(statisticEntry: StatisticEntryDeprecated): Expenses {
        val costDistribution = TestHelper.makeCostDistribution(statisticEntry.getEntryFor(TestHelper.thomas), 0, 0, statisticEntry.getEntryFor(TestHelper.thomas))
        return Expenses("", "", "", costDistribution.sumReal(), costDistribution, statisticEntry.date, "")
    }
}
